docs/site/src/components/ApiMenu.js

```diff
@@ -0,0 +1,68 @@
+// @flow weak
+
+import React, { PureComponent, PropTypes } from 'react';
+import { camelCase } from 'docs/site/src/utils/helpers';
+import IconButton from 'material-ui/IconButton';
+import { Menu, MenuItem } from 'material-ui/Menu';
+import MoreVertIcon from 'material-ui/svg-icons/more-vert';
+
+
+export default class ApiMenu extends PureComponent {
```

Comment:
My point is conceptual. `PureComponent` inherit from `Component` to implement the logic.
Inheritance has proven to be too power for keeping code independent. On the other hand, a functional composition provides a simple contract: `input -> output`.
At the office, my coworkers like writing components like this:
```js
export default recompose.compose(
  recompose.setDisplayName('PatientHistoryLoader'),
  recompose.connect(({patientAppointmentsTask$}) => ({
    show: patientAppointmentsTask$.pluck('progress'),
  })),
  recompose.branch(
    ({show}) => !show,
    recompose.renderNothing,
    recompose.identity,
  ),
  recompose.pure,
)(() => (
  <Loader className="booking-appointment-patient-history-loader" />
));
```